src/SchemaRegistry.ts

```diff
@@ -165,39 +166,50 @@ export default class SchemaRegistry {
   }
 
   private async updateOptionsWithSchemaReferences(
-    options: SchemaRegistryAPIClientOptions | undefined,
     schema: ConfluentSchema,
+    options?: SchemaRegistryAPIClientOptions,
   ) {
     const helper = helperTypeFromSchemaType(schema.type)
-    const referredSchemas = await this.getReferredSchemas(schema, helper)
-    return helper.updateOptionsFromSchemaReferences(options as ProtocolOptions, referredSchemas)
+    const referencedSchemas = await this.getreferencedSchemas(schema, helper)
+
+    const protocolOptions = this.asProtocolOptions(options)
+    return helper.updateOptionsFromSchemaReferences(referencedSchemas, protocolOptions)
   }
 
-  private async getReferredSchemas(
+  private asProtocolOptions(options?: SchemaRegistryAPIClientOptions): ProtocolOptions | undefined {
+    if (!(options as LegacyOptions)?.forSchemaOptions) {
+      return options as ProtocolOptions | undefined
+    }
+    return {
+      [SchemaType.AVRO]: (options as LegacyOptions)?.forSchemaOptions,
+    }
+  }
+
+  private async getreferencedSchemas(
     schema: ConfluentSchema,
     helper: SchemaHelper,
   ): Promise<ConfluentSchema[]> {
     const referencesSet = new Set<string>()
-    return this.getReferredSchemasRecursive(schema, helper, referencesSet)
+    return this.getreferencedSchemasRecursive(schema, helper, referencesSet)
   }
 
-  private async getReferredSchemasRecursive(
+  private async getreferencedSchemasRecursive(
     schema: ConfluentSchema,
     helper: SchemaHelper,
     referencesSet: Set<string>,
   ): Promise<ConfluentSchema[]> {
     const references = schema.references || []
 
-    let referredSchemas: ConfluentSchema[] = []
+    let referencedSchemas: ConfluentSchema[] = []
```

Comment:
```suggestion
    const referencedSchemas: ConfluentSchema[] = []
```

Reply:
This is not possible because `referencedSchemas` is updated

Reply:
See my related comment below - instead of re-declaring `referencedSchemas` and mutating it, you can simply use [.push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) to push to the array instead.

```javascript
referencedSchemas.push(...schemas);
```